docs/development/designs/automationGo.md

```diff
@@ -0,0 +1,49 @@
+# Problem
+
+Maintaining unsupported versions of Kubernetes and by extention Golang/Go requires large amounts of monitoring, evaluation, occassional deep dives into the functionality and differences in versions, and large amounts of repetitive searchable documentation for the Open-Source Community and transparency.
+
+Specifically for patching golang, we are required to monitor their announcements for vulnerabilities, open our own CVE issues for backports that are consistent to allow for easily searchable auditing and transparency for the open-source community, attempt backporting the fixes, and when the backport doesn't apply cleanly, discover the differences in versions to make the backport apply, if applicable.
+
+While the last task will require effort by the team, the rest of the task can be automated to provide clear and consistent [unknown phrasing/word here]. An automated solution would build a consistant audit trail for which CVEs have been patched or why they weren't, any public discussion on the decision process, and many other data points that could be of interest.
+
+# Solution Overview
+
+I propose creating a go application for patching. Starting with Golang and potentially extending the functionality to Kubernetes as the workflow should be similar.
+
+## Golang Patching Runbook
+
+1. Golang release CVE announcement
+2. Open cve backport issues for EKS supported versions
+3. Attempt cherry-pick
+   3a. Find upstream commit fix
+   3b. git cherry-pick {commit hash}
+   3c. fix any merge issues
```

Comment:
we'll need a good way to follow the attempted cherry pick in automation, as I imagine it'll often fail at this step. 

It seems like we could have a couple of options here....

- fail the job and post the failure status to the issue with a link to the job
- commit and generate the patch file anyway , proceeeding to step 5, and let a human engineer sort 'em out

I think I like option 1 -- if it fails, just notify on the issue, and a human engineer takes over from there

Reply:
I agree with the thought that option 1 would be the best course of action. I feel option 2 could have more oops moments or additional complications by trying to force create a patch.